pkg/rpc/codec.go

```diff
@@ -1,35 +1,62 @@
 package rpc
 
 import (
-	gogoproto "github.com/gogo/protobuf/proto"
-	"github.com/golang/protobuf/proto" //nolint:staticcheck
 	"google.golang.org/grpc/encoding"
+	"google.golang.org/grpc/mem"
 )
 
 const name = "proto"
 
-type codec struct{}
+type gogoprotoMessage interface {
+	MarshalToSizedBuffer([]byte) (int, error)
+	Unmarshal([]byte) error
+	ProtoSize() int
+}
+
+var pool = mem.DefaultBufferPool()
+
+type codec struct {
+	fallback encoding.CodecV2
+}
 
-var _ encoding.Codec = codec{}
+var _ encoding.CodecV2 = &codec{}
 
 func init() {
-	encoding.RegisterCodec(codec{})
+	encoding.RegisterCodecV2(&codec{
+		fallback: encoding.GetCodecV2(name),
```

Comment:
Does this mean that the previous codec for proto will be set as a fallback for the newly registered codec? Is the codec for proto always registered?

Reply:
Yes, gRPC registers the default codec for "proto". If you want to delve into it, see this [code](https://github.com/grpc/grpc-go/blob/7dbf12ef2e723feb316960e64072d550a159d1e6/encoding/proto/proto.go#L19-L37).